_source/_posts/2023-09-14-spring-boot-keycloak-kubernetes.md

```diff
@@ -0,0 +1,729 @@
+
+```
+application {
+  config {
+    baseName gateway
+    applicationType gateway
+```
+
+Find out the IP address with the following command:
+
+```shell
+gcloud compute addresses describe gateway-ip --global --format='value(address)'
+```
+
+Create a `kubernetes` folder at the root of the project, and run the generator:
+
+```shell
+mkdir kubernetes
+cd kubernetes
+jhipster k8s
+```
+
+Choose the following options when prompted:
+
+- Type of application: **Microservice application**
+- Root directory: **../**
+- Which applications? (select all)
+- Set up monitoring? **No**
+- Which applications with clustered databases? select **store**
+- Kubernetes namespace: **demo**
+- Docker repository name: **your-dockerhub-username**
+- Command to push Docker image: `docker push`
+- Enable Istio? **No**
+- Kubernetes service type? **Ingress**
+- Kubernetes ingress type? **Google Kubernetes Engine Ingress**
+- Root FQDN for ingress services: **\<public-ip\>.nip.io**
+- Use dynamic storage provisioning? **Yes**
+- Use a specific storage class? (leave empty)
+
+### Customize the configuration
+
+The generated Kubernetes descriptors will create a certificate issuer in the cluster, pointing to Let's Encrypt staging environment. The Let's Encrypt issuer requires an email address to remind you to renew the certificate after 30 days before expiry. You will only receive this email if something goes wrong when renewing the certificate with cert-manager. Set the email address in the issuer descriptor `kubernetes/cert-manager/letsencrypt-staging-issuer.yml`:
+
+```yml
+spec:
+  acme:
+    server: https://acme-staging-v02.api.letsencrypt.org/directory
+    email: <your-email>
+```
+
+### Deploy to Google Kubernetes Engine (GKE)
+
+Create a Kubernetes cluster on Google Cloud:
+
+```shell
+gcloud container clusters create jhipster-cluster \
+  --zone southamerica-east1-a \
+  --machine-type n1-standard-4 \
+  --enable-autorepair \
+  --enable-autoupgrade
+```
+
+**Note**: you can choose a different zone and machine type.
+
+Install the [gke-gcloud-auth-plugin](https://cloud.google.com/blog/products/containers-kubernetes/kubectl-auth-changes-in-gke):
+
+```shell
+gcloud components install gke-gcloud-auth-plugin
+```
+
+Then, fetch the cluster credentials with:
+
+```shell
+gcloud container clusters get-credentials jhipster-cluster --zone southamerica-east1-a
+```
+
+`get-credentials` will update a `kubeconfig` file with appropriate credentials and endpoint information to point `kubectl` at a specific cluster in Google Kubernetes Engine.
+
+Install [cert-manager](https://cert-manager.io/docs/tutorials/getting-started-with-cert-manager-on-google-kubernetes-engine-using-lets-encrypt-for-ingress-ssl/) in your cluster:
+
+```shell
+kubectl apply -f https://github.com/cert-manager/cert-manager/releases/download/v1.13.1/cert-manager.yaml
```

Comment:
I'm upgrading this to 1.13.2 because there are CVEs in 1.13.1. https://github.com/cert-manager/cert-manager/releases/tag/v1.13.2